corehq/blobs/s3db.py

```diff
@@ -60,12 +63,19 @@ def record_long_request(duration):
     def put(self, content, **blob_meta_args):
         meta = self.metadb.new(**blob_meta_args)
         check_safe_key(meta.key)
-        s3_bucket = self._s3_bucket(create=True)
+        if settings.BUCKET_NAME_FUNCTION:
+            s3_bucket_name = settings.BUCKET_NAME_FUNCTION()
+            s3_bucket = self._s3_bucket(bucket=s3_bucket_name)  # assumes bucket exists
```

Comment:
in that case you might as well bypass this function altogether since it's whole purpose is to create buckets if they don't exist. You could just do:
```suggestion
            self.db.Bucket(s3_bucket_name)
```

Alternately if the bucket's are being created outside of the blobdb then you could have the `BUCKET_NAME_FUNCTION` return an actual bucket instead of just the name.

Reply:
Ah. I meant to include a todo in that comment to make sure that it does create the buckets. 

From your above comment:

> I'm now thinking that it might make more sense to have the blobdb track which bucket's it has seen before 

It sounds like you think this approach is better than having `BUCKET_NAME_CLASS` or something similar create the buckets, correct?